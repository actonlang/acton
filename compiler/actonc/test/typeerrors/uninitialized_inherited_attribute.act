class Base(object):
    base_count: int
    
    def __init__(self):
        self.base_count = 0

class Derived(Base):
    derived_value: str
    
    def __init__(self):
        # Forgot to call Base.__init__ or initialize base_count
        self.derived_value = "hello"

actor main(env):
    d = Derived()
    env.exit(0)