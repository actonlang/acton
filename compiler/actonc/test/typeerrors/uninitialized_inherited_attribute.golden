Building file test/typeerrors/uninitialized_inherited_attribute.act using temporary scratch directory
  Compiling uninitialized_inherited_attribute.act for development
[error Compilation error]: attribute 'base_count' is not initialized in __init__
     +--> uninitialized_inherited_attribute.act@2:5-2:15
     |
   2 |     base_count: int
     :     ^--------- 
     :     `- attribute 'base_count' is not initialized in __init__
-----+
